pgmpy/models/MarkovChain.py

```diff
@@ -186,11 +186,23 @@ def add_transition_model(self, variable, transition_model):
         >>> model = MC()
         >>> model.add_variable('grade', 3)
         >>> grade_tm = {0: {0: 0.1, 1: 0.5, 2: 0.4}, 1: {0: 0.2, 1: 0.2, 2: 0.6 }, 2: {0: 0.7, 1: 0.15, 2: 0.15}}
+        >>> grade_tm_matrix = np.array([[0.1, 0.5, 0.4], [0.2, 0.2, 0.6], [0.7, 0.15, 0.15]])
         >>> model.add_transition_model('grade', grade_tm)
+        >>> model.add_transition_model('grade', grade_tm_matrix)
         """
         # check if the transition model is valid
-        if not isinstance(transition_model, dict):
-            raise ValueError('Transition model must be a dict.')
+        if not isinstance(transition_model, dict) and not isinstance(transition_model, np.ndarray):
+            raise ValueError('Transition model must be a dict or numpy array')
+        if isinstance(transition_model, np.ndarray) and len(transition_model.shape) != 2:
+            raise ValueError('Must be 2d array')
```

Comment:
@kris-singh IMO this error is a bit ambiguous. I think something like Transition model must be a 2d array like object should be better. The current error doesn't link it to transition model. Please change that.